camel-idea-plugin/src/main/java/org/apache/camel/idea/util/CamelIdeaUtils.java

```diff
@@ -180,6 +171,35 @@ public boolean acceptForAnnotatorOrInspection(PsiElement element) {
             .anyMatch(extension -> extension.acceptForAnnotatorOrInspection(element));
     }
 
+    public PsiClass getBean(PsiElement element) {
+        return enabledExtensions.stream()
+            .map(c -> c.getBeanClass(element))
+            .filter(Objects::nonNull)
+            .findFirst().orElse(null);
+    }
+
+    public PsiElement getBeanPsiElement(PsiElement element) {
+        return enabledExtensions.stream()
+            .map(c -> c.getBeanPsiElement(element))
+            .filter(Objects::nonNull)
+            .findFirst().orElse(null);
+    }
+
+    public boolean isAnnotatedWithHandler(PsiMethod psiMethod) {
+        return  Arrays.stream(psiMethod.getAnnotations()).anyMatch(a -> a.getQualifiedName().equals("org.apache.camel.Handler"));
```

Comment:
Cannon use class ref ?

Reply:
No because we don't depend on camel core in the plugin code, and second it can't be a hard reference because it will throw a lot of class not found exception if the user has none Camel project open in IDEA